src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrPusher.cs

```diff
@@ -0,0 +1,173 @@
+
+        string? pat = string.Empty;
+        if (repoType == GitRepoType.GitHub)
+        {
+            pat = _vmrRemoteConfiguration.GitHubToken;
+        }
+        if (repoType == GitRepoType.AzureDevOps)
+        {
+            pat = _vmrRemoteConfiguration.AzureDevOpsToken;
+        }
+
+        _localGitRepo.Push(
+            _vmrInfo.VmrPath,
+            branch,
+            remoteUrl,
+            pat,
+            new LibGit2Sharp.Identity(Constants.DarcBotName, Constants.DarcBotEmail));
+    }
+
+    private async Task<bool> CheckCommitAvailability(string gitHubApiPat, CancellationToken cancellationToken)
+    {
+        var commitsSearchArguments = _sourceManifest
+            .Repositories
+            .Select(r => (GitRepoUrlParser.GetRepoNameAndOwner(r.RemoteUri), r.CommitSha))
+            .Select(r => new CommitSearchArguments(r.Item1.RepoName, r.Item1.Org, r.CommitSha));
+
+        var commits = await GetGitHubCommits(commitsSearchArguments, gitHubApiPat, cancellationToken);
+        
+        if (commits == null || commits.Data == null)
+        {
+            _logger.LogError("The Graphql query is not in the correct format.");
+            return false;
+        }
+
+        foreach (var pair in commits.Data)
+        {
+            if (pair.Value == null)
+            {
+                var commitArgs = commitsSearchArguments.First(r => GetGraphQlIdentifier(r.RepoName) == pair.Key);
+                _logger.LogError($"Repository {Constants.GitHubUrlPrefix}{commitArgs.RepoOwner}/{commitArgs.RepoName} was not found!");
+                return false;
+            }
+
+            if (pair.Value.Object == null)
+            {
+                var commitArgs = commitsSearchArguments.First(r => GetGraphQlIdentifier(r.RepoName) == pair.Key);
+                _logger.LogError($"Commit {commitArgs.Sha} was not found in {Constants.GitHubUrlPrefix}{commitArgs.RepoOwner}/{commitArgs.RepoName}!");
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    /// Sends a query to GitHub GraphQL API searching for all commits in their respective public repos in one joined request
+    /// The result of each search is a property in the "data" object returned by the API
+    /// where Data.RepoName.Object.Id is a string if the commit was found in "RepoName"
+    /// and Data.RepoName.Object is null if the commit was not found
+    /// </summary>
+    private async Task<CommitsQueryResult?> GetGitHubCommits(
+        IEnumerable<CommitSearchArguments> commits,
+        string gitHubApiPat,
+        CancellationToken cancellationToken)
+    {
+        var queries = commits
+            .Select(c => $"{GetGraphQlIdentifier(c.RepoName)}: repository(name: \\\"{c.RepoName}\\\", owner: \\\"{c.RepoOwner}\\\"){{object(oid: \\\"{c.Sha}\\\"){{ ... on Commit {{id}}}}}}");
+        var query = string.Join(", ", queries);
+        var body = @"{""query"" : ""{" + query + @"}""}";
```

Comment:
I'd also probably use the same style here and remove `@`?

Reply:
sure